mclf/curves/smooth_projective_curves.py

```diff
@@ -1265,3 +1268,75 @@ def e_f_of_valuation(v):
     if hasattr(v, "_approximation"):
         v = v._approximation
     return (v.E(), v.F())
+
+
+def extension_of_finite_field(K, n):
+    r""" Return a field extension of this finite field of degree n.
+
+    INPUT:
+
+    - ``K`` -- a finite field
+    - ``n`` -- a positive integer
+
+    OUTPUT: a field extension of `K` of degree `n`.
+
+    This function is useful if `K` is constructed as an explicit extension
+    `K = K_0[x]/(f)`; then ``K.extension(n)`` is not implemented.
+
+    """
+    assert K.is_field()
+    assert K.is_finite()
+    q = K.order()
+    R = PolynomialRing(K, 'z'+str(n))
+    z = R.gen()
+    # we look for a small number e dividing q^n-1 but not q-1
+    e = min([d for d in (q**n-1).divisors() if not d.divides(q-1)])
+    F = (z**e-1).factor()
+    f = [g for g, e in F if g.degree() == n][0]
+    # this is very inefficient!
+    return K.extension(f, 'z'+str(e))
+
+
+def make_finite_field(k):
+    r""" Return the finite field isomorphic to this field.
+
+    INPUT:
+
+    - ``k`` -- a finite field
+
+    OUTPUT: a triple `(k_1,\phi,\psi)` where `k_1` is a 'true' finite field,
+    `\phi` is an isomorphism from `k` to `k_1` and `\psi` is the inverse of `\phi`.
+
+    This function is useful when `k` is constructed as a tower of extensions
+    with a finite field as a base field.
+
```

Comment:
I'd add a comment here that this function should be removed once trac's 26103 has been merged.

Reply:
Ok, I'll do that.